src/peripheral/tpiu.rs

```diff
@@ -29,3 +31,71 @@ pub struct RegisterBlock {
     /// TPIU Type
     pub _type: RO<u32>,
 }
+
+/// The available protocols for the trace output.
+pub enum TraceProtocol {
+    /// Parallel trace port mode
+    Parallel = 0b00,
+    /// Asynchronous SWO, using Manchester encoding
+    AsyncSWOManchester = 0b01,
+    /// Asynchronous SWO, using NRZ encoding
+    AsyncSWONRZ = 0b10,
+}
+
+/// The SWO options supported by the TPIU.
+#[allow(dead_code)]
+pub struct SWOSupports {
+    /// Whether UART/NRZ encoding is supported for SWO.
+    nrz_encoding: bool,
+    /// Whether Manchester encoding is supported for SWO.
+    manchester_encoding: bool,
+    /// Whether parallel trace port operation is supported.
+    parallel_operation: bool,
+    /// The minimum implemented FIFO queue size of the TPIU for trace data.
+    min_queue_size: u8,
+}
+
+impl TPIU {
+    /// Sets the prescaler value for a wanted baud rate of the Serial
+    /// Wire Output (SWO) in relation to a given asynchronous refernce
+    /// clock rate.
+    #[inline]
+    pub fn set_swo_baud_rate(&mut self, ref_clk_rate: u32, baud_rate: u32) {
```

Comment:
Is it possible to find the frequency of the asynchronous reference clock?

Reply:
For the STM32F401RE, in RM0368, on p. 834:
> By default, the TRACECLKIN input clock of the TPIU is tied to GND. It is assigned to HCLK
two clock cycles after the bit TRACE_IOEN has been set.

Finding the frequency is thus assumed to be platform/vendor-specific. Nothing that can be done in this crate. Refer to HALs.